### ISSUE TITLE
Replace HtBody, HtStatus and HtHead with a unique object for the whole HTTP message

### ISSUE THREAD

**Issue by victornoel** [yegor256/cactoos-http#29]
This is just an idea, but looking at #3 or #4, I can see that an HTTP message is meant to be a whole stream.

What if we had an object, `HtMessage`, that provides `head()`, `body()` and so on to take care of this fact? It would solve the performance problems that seem to be clearly interrelated as well as make things clearer about the reality of what an HTTP message is.

**Comment by 0crat** [comment]
@yegor256[/z](https://www.0crat.com/u/yegor256) please, pay attention to this issue

**Comment by llorllale** [comment]
@victornoel I think your abstraction is too high, and is definitely not the direction this project is going/has gone so far. Why not just think in terms of the stream itself? What you need is something that will read the stream only up to a certain point.

WDYT?

**Comment by victornoel** [comment]
> I think your abstraction is too high, and is definitely not the direction this project is going/has gone so far

Agree, that's my main concern with my proposal

> Why not just think in terms of the stream itself? What you need is something that will read the stream only up to a certain point.

So `HtStatus` reads up to the separator between status and headers, HtHead reads up to the separator between headers and body and skip the status and `HtBody` skips status and header and gives back the rest.

So #28 is one aspect of the solution then and should be complemented with something that skips until a certain series of bytes? If you agree let's close this and I will open an issue for a `SkippedInput` or whatever other name fits well.

**Comment by llorllale** [comment]
@victornoel no, let's go the same direction they went with in #28: `HtBody` will do the "skipping" on the stream. And then, in the future, if there is anything in particular to be extracted from a body, then it should operate off of `HtBody`'s stream.

**Comment by 0crat** [comment]
The job is not in WBS, won't close the order
